### PR TITLE
feat: Add blockDeviceMappings fields to support multiple EBS volumes

### PR DESCRIPTION
**Pull request by lindarr915**
BREAKING CHANGE: remove blockDevice fields in helm values

### CODE REVIEW

**Comment by lindarr915** [comment]
Relate to the issue https://github.com/aws-ia/terraform-aws-eks-data-addons/issues/40

File: helm-charts/karpenter-resources/values.yaml [hunk 1]
```diff
@@ -16,8 +16,8 @@ ec2NodeClass:
     httpProtocolIPv6: disabled
     httpPutResponseHopLimit: 2
     httpTokens: required
-  blockDevice:
-    deviceName: /dev/xvda
+  blockDeviceMappings:
+  - deviceName: /dev/xvda
     volumeSize: 100Gi
```
Comment (vara-bonthu) [inline helm-charts/karpenter-resources/values.yaml:21]: Do you need to add `ebs:` here? Check the docs here https://karpenter.sh/docs/concepts/nodeclasses/#specblockdevicemappings

**Review by vara-bonthu — COMMENTED** [review]
@lindarr915 Thanks, I have added one comment.  Please make sure this change has been tested locally in your environment. I normally deploy the helm chart changes by pointing to local folder path with the changes.

File: helm-charts/karpenter-resources/values.yaml (hunk 1 above)
Comment (lindarr915) [reply helm-charts/karpenter-resources/values.yaml:21]: Yes, I should have added ebs: here. My local test cases only covers overwriting the default values

File: helm-charts/karpenter-resources/Chart.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 apiVersion: v2
 name: karpenter-resources
 description: Helm chart for configuring custom resources for Karpenter on the cluster
-version: 0.0.1
-appVersion: 0.0.1
+version: 0.0.2
+appVersion: 0.0.2
```
Comment (vara-bonthu) [inline helm-charts/karpenter-resources/Chart.yaml:5]: Changing this version might not make a difference since we are not publishing these Helm charts. You can keep it as 0.0.1.

I will create a new tag for the Data Add-ons, so that existing blueprints will still work with the old tags, and the new ones will need updating to below.

```hcl
  source  = "aws-ia/eks-data-addons/aws"
  version = "~> 1.40" 
```

File: karpenter-resources.tf [hunk 3]
```diff
@@ -1,6 +1,6 @@
 locals {
   namespace = "karpenter-resources"
-  version   = "0.0.1"
+  version   = "0.0.2"
```
Comment (vara-bonthu) [inline karpenter-resources.tf:3]: you can revert this change to 0.0.1

File: helm-charts/karpenter-resources/Chart.yaml (hunk 2 above)
Comment (lindarr915) [reply helm-charts/karpenter-resources/Chart.yaml]: reverted back to 0.0.1

**Review by lusoal — APPROVED** [review]
LGTM

**Review by askulkarni2 — APPROVED** [review]
